### PR TITLE
TMEDIA-196 - Change imports to utilize the SDK version of constructSocialURL

### PR DESCRIPTION
**Pull request by nschubach**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
Removes old social link generating code and utilize the SDK version.

## Jira Ticket
- [TMEDIA-196](https://arcpublishing.atlassian.net/browse/TMEDIA-196)

## Acceptance Criteria
N/A

## Test Steps
- This implements code on the `author-bio-block` and `full-author-bio-block` blocks to handle returning the url for social links on the biography pages.  They will now utilize the shared SDK version of this code which is a near duplicate of the old implementation with some developer QOL improvements.  No visible impact should be noticed on biography pages and the social links should act as they had before.

1. Checkout this branch `git checkout TMEDIA-196-construct-social-media-links`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/author-bio-block,@wpmedia/full-author-bio-block`
3. Find a biography page ([localhost full-author-bio example](http://localhost/pagebuilder/editor/curate?p=p09JD0qqD50JRr&v=v059xCnqD50JRr)) and validate the social media links contained therein.

### Before
When clicking on the social media links it would direct you to the appropriate URL.

### After
When clicking on the social media links it would direct you to the appropriate URL as before.  No differences should be noted.

## Dependencies or Side Effects
Depends on: https://github.com/WPMedia/engine-theme-sdk/pull/314

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Review by JackHowa — CHANGES_REQUESTED** [review]
Looks like the mock is failing in the test action. if you go to the test, the function can be added to the jest mocks for author-bio default test https://github.com/WPMedia/fusion-news-theme-blocks/runs/2384591212?check_suite_focus=true#step:6:24. similar to this change, where I added the function to the engine theme sdk, then mocked it in the blocks https://github.com/WPMedia/fusion-news-theme-blocks/pull/835/files#diff-922e8ea6b9d0ae6055ca7090092839c642da3cec6853e2b6dbc55826ccd7e459R9

**Comment by nschubach** [comment]
@JackHowa I have corrected the failed mocks/tests in a way that I think is still suitable to perform the intent of the tests as they were written.

**Comment by JackHowa** [comment]
> @JackHowa I have corrected the failed mocks/tests in a way that I think is still suitable to perform the intent of the tests as they were written.

cool, I'll review this again!

File: blocks/full-author-bio-block/features/full-author-bio/default.test.jsx [hunk 1]
```diff
@@ -199,7 +200,7 @@ describe('the full author bio block', () => {
       it('should have an instagram url', () => {
```
Comment (JackHowa) [inline blocks/full-author-bio-block/features/full-author-bio/default.test.jsx]: we can probably delete this test. it's basically just testing the mock as far as I can see.

File: blocks/full-author-bio-block/features/full-author-bio/default.test.jsx [hunk 2]
```diff
@@ -199,7 +200,7 @@ describe('the full author bio block', () => {
...
         const wrapper = mount(<FullAuthorBio />);
 
-        expect((wrapper.find('.instagram').prop('href'))).toEqual('https://www.instagram.com/janedoe/');
+        expect((wrapper.find('.instagram').prop('href'))).toEqual('janedoe');
```
Comment (JackHowa) [inline blocks/full-author-bio-block/features/full-author-bio/default.test.jsx]: same with this one

File: blocks/author-bio-block/features/author-bio/default.test.jsx [hunk 3]
```diff
@@ -624,7 +625,7 @@ describe('Given the list of author(s) from the article', () => {
     const wrapper = mount(<AuthorBio />);
 
     const socialButtonsContainer = wrapper.find('section.socialButtons');
-    expect(socialButtonsContainer.find('a').props().href).toBe('mailto:bernstein@washpost.com');
+    expect(socialButtonsContainer.find('a').props().href).toBe('bernstein@washpost.com');
```
Comment (JackHowa) [inline blocks/author-bio-block/features/author-bio/default.test.jsx]: this line can probably be deleted. it mostly tests the mock formatURL right?

Comment (JackHowa) [inline blocks/author-bio-block/features/author-bio/default.test.jsx]:
```suggestion
```

File: blocks/full-author-bio-block/features/full-author-bio/shared/constructSocialURL.test.js [hunk 4]
```diff
@@ -1,45 +0,0 @@
-import constructSocialURL from './constructSocialURL';
```
Comment (JackHowa) [inline blocks/full-author-bio-block/features/full-author-bio/shared/constructSocialURL.test.js:1]: nice work

File: blocks/full-author-bio-block/features/full-author-bio/default.test.jsx [hunk 5]
```diff
@@ -26,6 +26,7 @@ jest.mock('@wpmedia/engine-theme-sdk', () => ({
   RssIcon: () => <svg>RssIcon</svg>,
   LazyLoad: ({ children }) => <>{ children }</>,
   isServerSide: () => true,
+  constructSocialURL: (type, field) => field,
```
Comment (JackHowa) [inline blocks/full-author-bio-block/features/full-author-bio/default.test.jsx:29]: good job adding the mock

**Review by JackHowa — CHANGES_REQUESTED** [review]
there were some extra tests that could probably be removed. if they're testing the mock, then we can remove it. sorry if this is tough to get on your first pr. I like to think about https://kentcdodds.com/blog/the-merits-of-mocking

**Comment by nschubach** [comment]
> sorry if this is tough to get on your first pr.

@JackHowa : Not even... I get it

**Review by JackHowa — APPROVED** [review]
approved! great work. good job keeping with your first one

**Comment by nschubach** [comment]
@JackHowa had to do a canary merge because of changes on the branch.

**Pull request merged**